builtin/docker/builder.go

```diff
@@ -284,6 +281,14 @@ func (b *Builder) Build(
 	return result, nil
 }
 
+func createKeyValuePairs(m map[string]string) string {
+	b := new(bytes.Buffer)
+	for key, value := range m {
+		fmt.Fprintf(b, "--build-arg %s=%s ", key, value)
```

Comment:
I would strongly suggest quoting the parameters (so that `ENV_NAME="some value"` doesn't break the build) and to use an array of strings + join instead of a buffer, so that you can easily join them:
```go
strings.Join([]string{"--build-arg x=y", "--build-arg a=b"}, " ")
```

In your case:
```go
b := []string{}
for key, value := range m {
    b = append(b, fmt.Fprintf(b, "--build-arg %s=%s", key, value))
}
return strings.Join(b, " ")
```